CONTRIBUTING.md

```diff
@@ -1,65 +1,189 @@
+   
+# 1. Introduction #
 
-2. Configure the project and build (CMake is our the preferred configuration system, but not the only one. More details in the [README.md](README.md)) :
-```
-mkdir build && cd build
-cmake .. && cmake --build . --config Release
-```
+This file describes how users and developers can contribute to the Exiv2 project. We welcome any help, like for example, contributing lens data (images), code contributions, and bug reports.
+In the past this was organized using Redmine at: [dev.exiv2.org](http://dev.exiv2.org). As of April 2017 the Exiv2 project has moved from svn and Redmine to GitHub and information, such
+as, Wikis etc., will gradually be moved from Redmine to GitHub. User requests on the Redmine forums and the Redmine issue tracking system will be serviced until Exiv2 0.27 is released. 
+After 0.27 is released the Redmine site will be read-only and all feature requests, bug reports etc., must be performed on GitHub. It is, however, preferred to use GitHub for all new 
+feature requests and bug reports to Exiv2.
 
-3. Make sure the tests pass:
-```
-make tests          # Application tests
-./bin/unit_tests    # Unit tests
-```
+# 2. Recommended Workflow for Contributing Code #
+
+Here we describe the recommended code contribution workflow after switching to GitHub. Code contributions is performed by using *pull requests* (PR) on GitHub. 
+For this to work one first needs to [create a user account on GitHub](https://help.github.com/articles/signing-up-for-a-new-github-account/) if one do not 
+already have one. A PR should preferable contain only one new feature/bug fix etc. Since it is not uncommon to work on several PRs at the same time 
+it is recommended to create a new _branch_ for each PR. In this way PRs can easily be separated and the review and merge process becomes cleaner.
+As a rule-of-thumb:
+
+- PRs should be kept at a manageable size. Try to focus in just one goal per PR. If you find yourself doing several things in a PR that were not expected, 
+then try to split the different tasks in different PRs.
+- Commits should always change a *single* logical unit so that cherry-picking & reverting is simple.
+- Commit messages should be as informative and concise as possible. The first line of the commit message should be < 80 characters and 
+  describe the commit briefly. If the 80 characters are too short for a summary, then consider splitting the commit. Alternatively, below the short summary 
+  add one blank line and then a more detailed explanation if required.
+
+See the [GIT_GUIDELINES.md](GIT_GUIDELINES.md) file for a more detailed description on the git workflow. 
+
+Below we outline the recommended steps in the code contribution workflow. We use `your-username` to refer to your username on GitHub, `exiv2_upstream` is used when we 
+set the upstream remote repository for Exiv2 (we could have picked any name except `origin` and `master` [TODO verify this]), and we use the name `my_first_pr_branch` 
```

Comment:
I think it is possible for some name duplicates in git, but I certainly wouldn't recommend doing that. Nevertheless, I would just remove the sentence in brackets, as there are also other branches & tags in exiv2 whose name you shouldn't use and listing them all will get a little too much.

Reply:
OK. I will remove the TODO. Perhaps I can word it a little bit different so that people are aware of possible name collisions.